src/mathf/mathf.ts

```diff
@@ -858,9 +858,11 @@ export class mathf {
    * @return {number} The interporalated value.
    */
   static lerp(value1: number, value2: number, amount: number): number {
```

Comment:
Yeah.......so this is definitely the correct traditional lerp

(in c#)
```c#
Lerp(float a, float b, float t) {
  return (1.0f - t) * a + b * t;
}
```

but....I can't remember but as commented out, there was some reason why I went with current calculations as I did go back and forth at some point.  

For this reason, I'm a bit worried about regression here given since many classes depend on this working as is.  Do you see any issues?  If not, although not ideal, I would almost ask to leave this as is and make a lerp2.

Reply:
It looks like this version wasn't used previously, the initial commit is:
```

  /**
   * Linear interpolate from start to end given amount.
   * ```ts
   *
   * ```
   * @param {number} value1 The low-end of the range to lerp.
   * @param {number} value2 The high-end of the range to lerp.
   * @param {number} amount A value between 0-1.  1 would essentially mean no lerp.
   */
  static lerp(value1: number, value2: number, amount: number) {
    amount = amount < 0 ? 0 : amount;
    amount = amount > 1 ? 1 : amount;
    return value1 + (value2 - value1) * amount;
  }
```

The only concern would be if we have something that depends on how it currently handles negatives, but I'm not sure if that's something we should support?

I'll add a couple more tests that cover the issue this addresses.

Reply:
Yeah I think if it passes the tests in mathf.spec.ts it might be okay then.  :)